x/stakeibc/keeper/keeper.go

```diff
@@ -140,3 +143,36 @@ func (k Keeper) GetConnectionId(ctx sdk.Context, portId string) (string, error)
 	}
 	return "", fmt.Errorf("portId %s has no associated connectionId", portId)
 }
+
+// helper to get what share of the curr epoch we're through
+func (k Keeper) GetStrideEpochElapsedShare(ctx sdk.Context) (sdk.Dec, error) {
+	epochType := epochstypes.STRIDE_EPOCH
+	epochTracker, found := k.GetEpochTracker(ctx, epochType)
+	if !found {
+		k.Logger(ctx).Error(fmt.Sprintf("Failed to get epoch tracker for %s", epochType))
+		return sdk.ZeroDec(), sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Failed to get epoch tracker for %s", epochType)
+	}
+	prevEpochStartTime := epochTracker.NextEpochStartTime - epochTracker.Duration
+	elapsedShare := sdk.NewDec(ctx.BlockTime().UnixNano() - prevEpochStartTime).Quo(sdk.NewDec(epochTracker.Duration))
```

Comment:
nit: is it worth adding a check that `ctx.BlockTime().UnixNano() > prevEpochStartTime && ctx.BlockTime().UnixNano() < epochTracker.NextEpochStartTime`

Reply:
added this check and a check that `elapsedShare \in (0,1)`